common/config/rush/.npmrc

```diff
@@ -1,2 +1,12 @@
+# Rush uses this file to configure the package registry, regardless of whether the
+# package manager is PNPM, NPM, or Yarn.  Prior to invoking the package manager,
+# Rush will always copy this file to the folder where installation is performed.
+# When NPM is the package manager, Rush works around NPM's mishandling of
+# undefined environment variables by deleting any lines that reference undefined
+# environment variables.
+#
+# DO NOT SPECIFY AUTHENTICATION CREDENTIALS IN THIS FILE.  It should only be used
+# to configure registry sources.
+
 registry=https://registry.npmjs.org/
 always-auth=false

```

Comment:
It is better not having this line. It should be with auth config.

Reply:
This is what has worked in web-build-tools for a very long time.    NPM's default is false.

If I remember right, in the past setting always-auth to true  could cause "npm install" to fail sometimes when using the public NPM registry.  This could happen accidentally when people set it to true in their `~/.npmrc` file (e.g. for working with a private registry), so this line restored the NPM default.

I just confirmed that `rush install` works when `always-auth=true`.  However I'm hesitant to change this value to be different from NPM's default unless you have a compelling justification.

---
In reply to: [217213746](https://github.com/Microsoft/rush-example/pull/2#discussion_r217213746) [](ancestors = 217213746)